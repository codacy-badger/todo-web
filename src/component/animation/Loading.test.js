import React from "react";
import Loading from "component/animation/Loading";
import renderer from "react-test-renderer";

const TEST_LOADING_COMPLETE_HTML = <div>LOADING COMPLETED</div>;

describe("Loading component", () => {
  test("should render completed Loading", () => {
    const component = renderer
      .create(<Loading>{TEST_LOADING_COMPLETE_HTML}</Loading>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should render Loading in progress", () => {
    const component = renderer
      .create(<Loading loading={true}>{TEST_LOADING_COMPLETE_HTML}</Loading>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should render Loading with error", () => {
    const component = renderer
      .create(<Loading error={true}>{TEST_LOADING_COMPLETE_HTML}</Loading>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
