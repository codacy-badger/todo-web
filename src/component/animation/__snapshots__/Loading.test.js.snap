// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render Loading in progress 1`] = `
<div
  aria-label="animation"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "height": "400px",
      "margin": "0 auto",
      "marginTop": "100px",
      "outline": "none",
      "overflow": "hidden",
      "width": "400px",
    }
  }
  tabIndex="0"
  title=""
/>
`;

exports[`Loading component should render Loading with error 1`] = `
<div
  aria-label="animation"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "height": "300px",
      "margin": "0 auto",
      "marginTop": "100px",
      "outline": "none",
      "overflow": "hidden",
      "width": "400px",
    }
  }
  tabIndex="0"
  title=""
/>
`;

exports[`Loading component should render completed Loading 1`] = `
<div
  className="loading-component complete"
>
  <div>
    LOADING COMPLETED
  </div>
</div>
`;
